Library/Homebrew/formula.rb

```diff
@@ -1273,11 +1273,11 @@ def patch
   # Yields |self,staging| with current working directory set to the uncompressed tarball
   # where staging is a {Mktemp} staging context.
   # @private
-  def brew(fetch: true, keep_tmp: false, interactive: false)
+  def brew(fetch: true, keep_tmp: false, debug_symbols: false, interactive: false)
     @prefix_returns_versioned_prefix = true
     active_spec.fetch if fetch
     stage(interactive: interactive) do |staging|
-      staging.retain! if keep_tmp
+      staging.retain! if keep_tmp || debug_symbols
```

Comment:
As @carlocab has mentioned: I don't think this will be enough as it's wiped on reboots. The staging directory will need to change to be in e.g. `HOMEBREW_CACHE` or even somewhere in `HOMEBREW_PREFIX` for this to not get wiped.

Reply:
In order to do this change, what will be needed in `formula_installer`? Specifically the sandbox in the `build` function makes me think there's a bit more going on than I'm aware of. See here: https://github.com/Homebrew/brew/blob/0815a4481adcb276fa3cdcad62710f02955b6944/Library/Homebrew/formula_installer.rb#L909-L925.
Any ideas would be super helpful!

Reply:
I'd rather avoid touching the sandbox, so let's stick to the locations you can write to.
```ruby
sandbox.allow_write_temp_and_cache 
```
suggests you should be able to write to `HOMEBREW_CACHE`, so maybe we should go there?

Reply:
Done.